src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs

```diff
@@ -28,8 +27,8 @@ protected override UInt256 GetKeyForItem((UInt256, DateTime) item)
         }
 
         private readonly PendingKnownHashesCollection pendingKnownHashes = new PendingKnownHashesCollection();
-        private readonly HashSetCache<UInt256> knownHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 5);
-        private readonly HashSetCache<UInt256> sentHashes = new HashSetCache<UInt256>(Blockchain.Singleton.MemPool.Capacity * 2 / 5);
+        private readonly HashSetCache<UInt256> knownHashes = new HashSetCache<UInt256>(ProtocolSettings.Default.MemoryPoolMaxTransactions * 2 / 5);
```

Comment:
I do not think that this is necessarily a `ProtocolSettings` but rather a node configuration.
There should be other place to set these type of parameters that are not mandatory to be the same.

Reply:
This is not a problem to be solved by this PR.